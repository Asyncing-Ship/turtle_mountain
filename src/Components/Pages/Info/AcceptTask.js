import React, { Fragment } from "react";
import {
  Button,
  Box,
  Tag,
  TagIcon,
  TagLabel,
  ButtonGroup,
} from "@chakra-ui/core";
import { connect } from "react-redux";
import { withRouter } from "react-router";
import moment from "moment";
class AcceptTask extends React.Component {
  state = {
    accepted: false,
    task: { first_name: "Avg", last_name: "Joe", content: "f" },
    tagged_users: [
      { is_admin: true, first_name: "ex", last_name: "ample" },
      { is_admin: false, first_name: "non", last_name: "admin" },
    ],
  };
  render() {
    return (
      <>
        <h3>{this.state.task.content}</h3>

        <Box flex="1" textAlign="left">
          <small>
            <i>
              Posted at:{" "}
              {moment(this.state.task.date_posted).format("MM/DD/YY LT")} (By{" "}
              <u>
                {this.state.task.first_name} {this.state.task.last_name}
              </u>
              )
            </i>
          </small>
        </Box>
        <Box my={3}>
          {this.state.tagged_users.map((x, i) => (
            <Fragment key={i}>
              {x.is_admin ? (
                <Tag rounded="full" size="sm" variantColor="purple" mr={2}>
                  <TagIcon icon="star" size="10px" />
                  <TagLabel>
                    {x.first_name} {x.last_name}
                  </TagLabel>
                </Tag>
              ) : (
                <Tag rounded="full" size="sm" variantColor="yellow" mr={2}>
                  <TagLabel>
                    {x.first_name} {x.last_name}
                  </TagLabel>
                </Tag>
              )}
            </Fragment>
          ))}
        </Box>
        <Box textAlign="right" borderTop="1px solid #f5fffe" pt={3} mt={3}>
          <ButtonGroup>
            <Button
              size="sm"
              rightIcon="check"
              variantColor="green"
              className="new_class_goes_here"
              onClick={async () => {
                this.setState({ accepted: true });
              }}
            >
              Accept Task
            </Button>
          </ButtonGroup>
        </Box>
      </>
    );
  }
}

export default AcceptTask;
