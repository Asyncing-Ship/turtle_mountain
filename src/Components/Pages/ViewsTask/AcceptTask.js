import React, { Component } from "react";
import { Button } from "@chakra-ui/core";
import { connect } from "react-redux";

class AcceptTask extends Component {
  render() {
    return (
      <>
        <h3>{this.props.task.added_by}</h3>
        <h2>{this.props.task.title}</h2>
        <h3>{this.props.task.content}</h3>
        <Button
          onClick={async () => {
            await this.props.dispatch({
              type: "UPDATE_TASK",
              payload: {
                task_id: this.props.task.id,
                assigned_to: this.props.user.id,
              },
            });
            await this.props.history.push("/tasks");
          }}
        >
          Accept Task
        </Button>
      </>
    );
  }
}
const mapStateToProps = (state) => {
  return { user: this.state.user };
};
export default connect(mapStateToProps)(AcceptTask);
