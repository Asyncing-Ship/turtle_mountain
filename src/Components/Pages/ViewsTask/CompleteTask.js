import React, { Component } from 'react';

class CompleteTask extends Component {
  render() {
    return (
      <>
        Complete selected task here:
      </>
    )
  }
}

export default CompleteTask;