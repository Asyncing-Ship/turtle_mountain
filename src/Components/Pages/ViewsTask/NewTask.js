import React, { Component } from "react";
import {
  Input,
  Button,
  FormControl,
  FormLabel,
  Box,
  Heading,
  Textarea,
} from "@chakra-ui/core";

import { connect } from "react-redux";
import { withRouter } from "react-router";
import Select from "react-select";

class NewTask extends Component {
  state = {
    title: "",
    content: "",
    select: [],
  };

  componentDidMount() {
    this.props.dispatch({ type: "FETCH_ALL_USERS" });
  }
  handleChange = (event, value) => {
    this.setState({
      [value]: event.target.value,
    });
  };

  handleInputChange = async (event) => {
    await console.log(event.value);
    if (this.state.select.length < 8) {
      await this.setState({ select: [...this.state.select, event.value] });
    }
    await console.log(this.state.select);
  };
  render() {
    return (
      <form
        onSubmit={async (event) => {
          event.preventDefault();
          await this.props.dispatch({
            type: "ADD_TASK",
            payload: {
              title: this.state.title,
              content: this.state.content,
              user_ids: this.state.select.map((x) => x.id),
            },
          });
          await this.props.history.push("/open");
        }}
      >
        <FormControl textAlign="left" bg="#2f2e2e" p={5} rounded="lg">
          <Heading color="#f5fffe">New Task</Heading>
          <FormLabel htmlFor="task-title">Task Title</FormLabel>
          <Input
            _focus={{ bg: "#f5fffe", border: "2px solid #3182ce" }}
            autoComplete="off"
            id="task-title"
            aria-required="true"
            placeholder="Task Title"
            onChange={(event) => this.handleChange(event, "title")}
            value={this.state.title}
            variant="filled"
            mb={5}
            isRequired
          />
          <FormLabel htmlFor="task-body">Description</FormLabel>
          <Textarea
            _focus={{ bg: "#f5fffe", border: "2px solid #3182ce" }}
            id="task-body"
            placeholder="Describe the task..."
            onChange={(event) => this.handleChange(event, "content")}
            value={this.state.content}
            variant="filled"
            resize="vertical"
            mb={5}
            isRequired
          />
          <Box style={{ backgroundColor: "white" }} mb={5}>
            TAGGED USERS
            {this.state.select.map((x) => (
              <Box>{"@" + x.first_name + " " + x.last_name}</Box>
            ))}
          </Box>
          <small style={{ color: "white" }}>Select User(s) to tag</small>
          <Select
            placeholder="SELECT A USER"
            className="col-12 col-lg-3"
            defaultValue={0}
            options={this.props.users
              .filter((x) => {
                for (let user of this.state.select) {
                  if (user === x) {
                    return false;
                  }
                }
                return true;
              })
              .map((x) => {
                return {
                  label: x.first_name + " " + x.last_name,
                  value: x,
                  key: x.id,
                };
              })}
            onChange={(event) => {
              this.handleInputChange(event);
            }}
          ></Select>
          <Box textAlign="right">
            <Button type="submit" rightIcon="add" variantColor="green">
              Add Task
            </Button>
          </Box>
        </FormControl>
      </form>
    );
  }
}

const mapStateToProps = (state) => {
  return { user: state.user, users: state.users };
};

export default withRouter(connect(mapStateToProps)(NewTask));
