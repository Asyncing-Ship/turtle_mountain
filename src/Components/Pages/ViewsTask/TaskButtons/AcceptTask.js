import React from "react";
import { Button, useToast, Box } from "@chakra-ui/core";
import { connect } from "react-redux";
import { withRouter } from "react-router";
import EditTask from "./EditTask";
import DeleteTask from "./DeleteTask";

const AcceptTask = (props) => {
  const toast = useToast();

  if (props.task.user_id === props.user.id) {
    return (
      <div>
        <h3>{props.task.content}</h3>

        <Box textAlign="right">
          <Button
            size="sm"
            rightIcon="check"
            variantColor="green"
            className="new_class_goes_here"
            onClick={async () => {
              await toast({
                title: "Task accepted.",
                description: "You accepted this task",
                status: "success",
                duration: 5000,
                isClosable: true,
              });
              await props.dispatch({
                type: "ACCEPT_TASK",
                payload: {
                  task_id: props.task.id,
                },
              });
              await props.history.push("/my");
            }}
          >
            Accept Task
          </Button>
          <EditTask task={props.task} />
          <DeleteTask task={props.task} />
        </Box>
        {props.tags[0] &&
          props.tags[0].tagged_users &&
          props.tags[0].tagged_users.map((x) => <span>@{x} </span>)}
        <Box>
          {props.tags.map((x) => (
            <span>
              @{x.first_name} {x.last_name}{" "}
            </span>
          ))}
        </Box>
      </div>
    );
  } else {
    return (
      <div>
        <h3>{props.task.content}</h3>

        <Box textAlign="right">
          <Button
            size="sm"
            rightIcon="check"
            variantColor="green"
            className="new_class_goes_here"
            onClick={async () => {
              await toast({
                title: "Task accepted.",
                description: "You accepted this task",
                status: "success",
                duration: 5000,
                isClosable: true,
              });
              await props.dispatch({
                type: "ACCEPT_TASK",
                payload: {
                  task_id: props.task.id,
                },
              });
              await props.history.push("/my");
            }}
          >
            Accept Task
          </Button>
        </Box>
        {props.tags[0] &&
          props.tags[0].tagged_users &&
          props.tags[0].tagged_users.map((x) => <span>@{x} </span>)}
        <Box>
          {props.tags.map((x) => (
            <span>
              @{x.first_name} {x.last_name}{" "}
            </span>
          ))}
        </Box>
      </div>
    );
  }
};

const mapStateToProps = (state) => {
  return { user: state.user, tags: state.tasks.taskTags };
};

export default withRouter(connect(mapStateToProps)(AcceptTask));
