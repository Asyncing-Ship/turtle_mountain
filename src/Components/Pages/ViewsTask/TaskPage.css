.accordionHead {
  background-color: #c79e61;
  color: white;
}
.accordionHead:hover {
  background-color: #c79e61;
  color: black;
}
.accordionHead:active {
  color: black;
}
