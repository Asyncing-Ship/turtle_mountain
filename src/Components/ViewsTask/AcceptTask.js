import React, { Component } from 'react';

class AcceptTask extends Component {
  render() {
    return (
      <>
        Accept selected task here:
      </>
    )
  }
}

export default AcceptTask;