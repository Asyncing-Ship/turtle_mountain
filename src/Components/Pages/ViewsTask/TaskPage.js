// ----- Start of imports -----
// React Import:
import React, { Component } from "react";
// React Redux Imports:
import { connect } from "react-redux";
// Chakra UI Imports:
import {
  Accordion,
  AccordionHeader,
  AccordionIcon,
  AccordionItem,
  AccordionPanel,
  Box,
  Button,
  Input,
} from "@chakra-ui/core";
// React Router DOM Imports:
import {
  withRouter,
  HashRouter as Router,
  Link,
  Route,
} from "react-router-dom";
// Components Imports:
import NewTask from "./NewTask";
import AcceptTask from "./AcceptTask";
import "./TaskPage.css";
import CompleteTask from "./CompleteTask";

// ----- This is the TaskPage component -----
// This component is tha main page component for tasks that displays the task for the users.
export class TaskPage extends Component {
  // Local State:
  state = {
    selectedTask: 1,
    searchString: "",
  };

  componentDidMount() {
    this.props.dispatch({ type: "FETCH_TASKS" });
  }

  // Render function:
  render() {
    return (
      <Router>
        {/* 2 Buttons to filter task by status and assignment. the first 2 buttons will clear the search strings */}
        <Button
          m={3}
          onClick={() => this.setState({ selectedTask: 1, searchString: "" })}
        >
          Open Tasks
        </Button>
        <Button
          m={3}
          onClick={() => this.setState({ selectedTask: 2, searchString: "" })}
        >
          My Tasks
        </Button>
        {/* The input will be used to filter the tasks by matching results. it should do this without a button */}
        <Input
          m={3}
          value={this.state.searchString}
          onChange={(event) =>
            this.setState({ selectedTask: 3, searchString: event.target.value })
          }
        />
        {/* the first button will give us this result,
         after filtering by open tasks, we map each task to an accordion item with the title being the task title. 
         then feed each task into its own acceptTask component*/}
        {this.state.selectedTask === 1 && (
          <Accordion allowMultiple>
            {this.props.tasks
              .filter((x) => x.status === "open")
              .map((x) => (
                <AccordionItem defaultIsOpen="False">
                  {({ isExpanded }) => (
                    <>
                      <AccordionHeader
                        _expanded={{ bg: "#c79e61", color: "white" }}
                        _hover={{ bg: "#c79e61", color: "white" }}
                      >
                        <Box flex="1" textAlign="left">
                          {x.title}
                        </Box>
                        <AccordionIcon />
                      </AccordionHeader>
                      <AccordionPanel pb={4}>
                        <AcceptTask task={x} />
                      </AccordionPanel>
                    </>
                  )}
                </AccordionItem>
              ))}
          </Accordion>
        )}
        {/* the second button will give us this result,
         after filtering by tasks, assigned to the current user,
          we map each task to an accordion item with the title being the task title. 
         then feed each task into its own completeTask component*/}
        {this.state.selectedTask === 2 && (
          <Accordion allowMultiple>
            {this.props.tasks
              .filter((x) => {
                console.log(x.assigned_to, this.props.user.id);
                return x.assigned_to === this.props.user.id;
              })
              .map((x) => (
                <AccordionItem defaultIsOpen="False">
                  {({ isExpanded }) => (
                    <>
                      <AccordionHeader
                        _expanded={{ bg: "#c79e61", color: "white" }}
                        _hover={{ bg: "#c79e61", color: "white" }}
                      >
                        <Box flex="1" textAlign="left">
                          {x.title}
                        </Box>
                        <AccordionIcon />
                      </AccordionHeader>
                      <AccordionPanel className="apanel" pb={4}>
                        <CompleteTask task={x} />
                      </AccordionPanel>
                    </>
                  )}
                </AccordionItem>
              ))}
          </Accordion>
        )}
        {/* the input will give us this result,
         after filtering search string, we map 
         each task to an accordion item with the
          title being the task title. and the body being the content, 
          followed by the status of the task*/}
        {this.state.selectedTask === 3 && (
          <Accordion allowMultiple>
            {this.props.tasks
              .filter(
                (x) =>
                  x.title.includes(this.state.searchString) ||
                  x.content.includes(this.state.searchString) ||
                  x.first_name.includes(this.state.searchString) ||
                  x.last_name.includes(this.state.searchString)
              )
              .map((x) => (
                <AccordionItem defaultIsOpen="False">
                  {({ isExpanded }) => (
                    <>
                      <AccordionHeader
                        _expanded={{ bg: "#c79e61", color: "white" }}
                        _hover={{ bg: "#c79e61", color: "white" }}
                      >
                        <Box flex="1" textAlign="left">
                          {x.title}
                        </Box>
                        <AccordionIcon />
                      </AccordionHeader>
                      <AccordionPanel pb={4}>
                        <AcceptTask task={x} />
                      </AccordionPanel>
                    </>
                  )}
                </AccordionItem>
              ))}
          </Accordion>
        )}
        {/* a button that links to the new tasks page */}
        <Link to="/tasks/newTask">
          <Button>New Task</Button>
        </Link>
      </Router>
    );
  }
}
// ----- End of TaskPage component -----

// ----- Start of mapStateToProps function -----
const mapStateToProps = (state) => {
  return {
    tasks: state.tasks.tasks,
    user: state.user,
  };
};
// ----- End of mapStateToProps function -----

// ----- Default export of TaskPage component with router connection, Redux connection that maps to props.

export default withRouter(connect(mapStateToProps)(TaskPage));
