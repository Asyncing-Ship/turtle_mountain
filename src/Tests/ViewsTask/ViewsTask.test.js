import React from "react";
import { shallow, mount } from "enzyme";
import AcceptTask from "../../Components/Pages/ViewsTask/AcceptTask";
import CompleteTask from "../../Components/Pages/ViewsTask/CompleteTask";
import NewTask from "../../Components/Pages/ViewsTask/NewTask";
import TaskPage from "../../Components/Pages/ViewsTask/TaskPage";
import toJson from "enzyme-to-json";
import { createStore } from "redux";
import rootReducer from "../../Redux/Reducers/reducers";
import { NavLink } from "react-router-dom";
import { Button } from "@chakra-ui/core";
const store = createStore(rootReducer);
it("renders correctly", () => {
  shallow(<AcceptTask />);
});

it("renders correctly", () => {
  shallow(<CompleteTask />);
});
it("renders correctly", () => {
  shallow(<NewTask />);
});
it("renders correctly", () => {
  shallow(<TaskPage />);
});
