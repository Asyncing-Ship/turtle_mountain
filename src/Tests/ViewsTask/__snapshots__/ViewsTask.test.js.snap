// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<Fragment>
  Accept selected task here:
</Fragment>
`;

exports[`renders without crashing 2`] = `
<Fragment>
  Complete selected task here:
</Fragment>
`;

exports[`renders without crashing 3`] = `
<Fragment>
  Create new task here:
</Fragment>
`;

exports[`renders without crashing 4`] = `
<Fragment>
  This is the task page!
</Fragment>
`;
